src/main/java/com/piekoszek/nowaksharedrent/jwt/JwtService.java

```diff
@@ -1,8 +1,13 @@
 package com.piekoszek.nowaksharedrent.jwt;
 
+import com.piekoszek.nowaksharedrent.dto.User;
+
+import java.util.Date;
+
 public interface JwtService {
 
-    String generateToken(JwtData jwtData);
+    String generateToken(User user);
+    String generateToken (User user, Date exp);
```

Comment:
I think it would be safer not to have method to create token with any exp date, I'd change it to updateTokenData(JwtData jwtData), also try to have as little code duplication as possible in implementation.

Reply:
If I want to update apartment list which is in DB I have to add something more than JwtData as all User object logic is in AuthService. I made some changes. Please take a look if it's better now.